server.go

```diff
@@ -669,10 +726,35 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 
 		tlsManager: tlsManager,
 
+		channelPeers:        make(map[string]struct{}),
```

Comment:
If you agree with my previous suggestion about returning a single map from the db query, then imo these two maps could also be merged into a single one, essentially tracking the channel counts by state per peer.

Reply:
merged